mp4parse/src/unstable.rs

```diff
@@ -161,9 +161,9 @@ pub fn create_sample_table(
     // With large numbers of samples, the cost of many allocations dominates,
     // so it's worth iterating twice to allocate sample_table just once.
     let total_sample_count = sample_to_chunk_iter(&stsc.samples, &stco.offsets)
-        .by_ref()
-        .map(|(_, sample_counts)| sample_counts.to_usize())
-        .sum();
+        .try_fold(0usize, |acc, (_, sample_counts)| {
+            acc.checked_add(sample_counts.to_usize())
+        })?;
```

Comment:
Good catch on the missing checked add!

Because I can't resist a good FP chain, I think this can be made just slightly more readable with the following refactor:
```suggestion
        .map(|(_, sample_counts)| sample_counts.to_usize())
        .try_fold(0, usize::checked_add)?;
```

Reply:
That's tidier, thanks!